File: week1-array-string/problem01-배열두배만들기/heetae.py
# 문제: 배열 두 배 만들기
# 접근 방식: 리스트 컴프리헨션으로 모든 원소에 2를 곱한 새 배열을 만든다.
# 시간복잡도: O(n) (n은 numbers의 길이)

def solution(numbers):
    return [i*2 for i in numbers]

File: week3-hash-sort/problem09-가장큰수/heetae.java
// 문제: 가장 큰 수
// 접근 방식: 자바의 소트를 이용해서 비교한뒤 스트림으로 합산
// 시간복잡도: O(nlogn)

import java.util.*;
import java.util.stream.*;
class Solution {
    public String solution(int[] numbers) {
        String answer = Arrays.stream(numbers)
            .mapToObj(String::valueOf)
            .sorted((a,b)->(b+a).compareTo(a+b))
            .collect(Collectors.joining());
        
        if(answer.startsWith("0")) return "0";
        
        return answer;
    }
}
